Source/Microsoft.Teams.Apps.CompanyCommunicator.Prep.Func/PreparingToSend/GetRecipientDataBatches/GetRecipientDataListForTeamsActivity.cs

```diff
@@ -0,0 +1,102 @@
+        {
+            this.teamDataRepository = teamDataRepository;
+            this.sentNotificationDataRepository = sentNotificationDataRepository;
+        }
+
+        /// <summary>
+        /// Run the activity.
+        /// </summary>
+        /// <param name="context">Durable orchestration context.</param>
+        /// <param name="notificationDataEntity">Notification data entity.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task RunAsync(
+            DurableOrchestrationContext context,
+            NotificationDataEntity notificationDataEntity)
+        {
+            if (notificationDataEntity.Teams == null || notificationDataEntity.Teams.Count() == 0)
+            {
+                throw new InvalidOperationException("NotificationDataEntity's Teams property value is null or empty!");
+            }
+
+            await context.CallActivityWithRetryAsync<IEnumerable<UserDataEntity>>(
+                nameof(GetRecipientDataListForTeamsActivity.GetTeamRecipientDataListAsync),
+                new RetryOptions(TimeSpan.FromSeconds(5), 3),
+                notificationDataEntity);
+        }
+
+        /// <summary>
+        /// This method represents the "get recipient data list for teams" durable activity.
+        /// 1). It gets recipient data list for teams ("team general channels").
+        /// 2). Initialize sent notification data in the table storage.
+        /// </summary>
+        /// <param name="notificationDataEntity">Notification data entity.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        [FunctionName(nameof(GetTeamRecipientDataListAsync))]
+        public async Task GetTeamRecipientDataListAsync(
+            [ActivityTrigger] NotificationDataEntity notificationDataEntity)
+        {
+            var teamsRecipientDataList =
+                await this.GetTeamsRecipientDataEntityListAsync(notificationDataEntity.Teams);
+
+            await this.sentNotificationDataRepository
+                .InitializeSentNotificationDataForRecipientBatchAsync(notificationDataEntity.Id, teamsRecipientDataList);
+        }
+
+        /// <summary>
+        /// Get teams' recipient data entity list.
+        /// </summary>
+        /// <param name="teamIds">Team IDs.</param>
+        /// <returns>List of recipient data entity (user data entities).</returns>
+        private async Task<List<UserDataEntity>> GetTeamsRecipientDataEntityListAsync(IEnumerable<string> teamIds)
+        {
+            var teamDataEntities = await this.teamDataRepository.GetTeamDataEntitiesByIdsAsync(teamIds);
+
+            var teamReceiverEntities = new List<UserDataEntity>();
+
+            foreach (var teamDataEntity in teamDataEntities)
+            {
+                teamReceiverEntities.Add(
+                    new UserDataEntity
+                    {
+                        AadId = teamDataEntity.TeamId,
```

Comment:
This doesn't look right, especially in light of line 94 below.
TeamId is the 19:xxx ID, right? It can't be the AadId then.